Lab1/Lab1/Program.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+
+//variant 3
+namespace Lab1
+{
+    class Program
+    {
+        private static string _nick;
+        private static EndPoint _remotePoint;
+        private static readonly Socket Socket = new (AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
+        private static readonly List<(DateTime, string)> Messages = new();
+
+        private static void StartPage()
+        {
+            try
+            {
+                Console.Clear();
+                Console.WriteLine("Enter your nickname!");
+                _nick = Console.ReadLine();
+                Console.Clear();
+                Console.WriteLine("Hello, " + _nick);
+                Console.WriteLine("Enter your port");
+                int myPort = Convert.ToInt32(Console.ReadLine());
+                Socket.Bind(new IPEndPoint(IPAddress.Parse("127.0.0.1"), myPort));
+                Console.WriteLine("Enter ip of your companion");
+                var ip = Console.ReadLine();
+                Console.WriteLine("Enter port of your companion");
+                var port = Convert.ToInt32(Console.ReadLine());
+                _remotePoint = new IPEndPoint(IPAddress.Parse(ip), port);
+                Console.Clear();
+            }
+            catch
+            {
+                Error("Wrong format");
+                Thread.Sleep(1500);
+                StartPage();
+            }
+        }
+
+        private static void ReceiveMessages()
+        { 
+            try
+            {
+                while (true)
+                {
+                    StringBuilder value = new ();
+                    byte[] data = new byte[256];
+                    do
+                    {
+                        var bytes = Socket.ReceiveFrom(data, ref _remotePoint);
+                        value.Append(Encoding.Unicode.GetString(data, 0, bytes));
+                    }
+                    while (Socket.Available > 0);
+
+                    switch (value[0])
+                    {
+                        case '/':
```

Comment:
It would be much better if you extract all this magic strings into enums or constants at least.